enf-reducer/promote-common-constraints.metta

```diff
@@ -0,0 +1,42 @@
+(= (promoteCommonLiterals $expr)
+     (case (car-atom $expr)
+        (
+                (AND
+                    (let* 
+                        (
+                        (($gSet $children) (getGsetAndChildren $expr))
+                        ($result (processOrChildren (car-atom $children)))
+                        (($head $tail) $result)                     
+                        ($updatedExpr (concatTuple $gSet $head))
+                        ($final (concatTuple $updatedExpr ($tail)))                       
+                        )   
+                        (cons-atom AND $final)
+                    )
+            )
+            ($else $else)
+        )
+    )
+)
+
+(= (processOrChildren $expr)
+    (let* 
+        (
+            (($literals $children) (getGsetAndChildren $expr)) 
+            ($childrenLiteral   (collapse (getGsetAndChildren (superpose $children))))
```

Comment:
The `$childrenLiteral` is a tuple of the `guardSet`, if any,  and `children` right?